events: Missed loop optimization

I'm seeing the following warning when building the latest master (commit 43fcfbe309b9153ec6901c1c8dbedc7dfc406aac):
```
events.c: In function ‘janus_events_edit_events_mask’:
events.c:349:12: warning: missed loop optimization, the loop counter may overflow [-Wunsafe-loop-optimizations]
       while(ev) {
            ^
events.c: In function ‘janus_events_type_to_label’:
events.c:371:7: warning: missed loop optimization, the loop counter may overflow [-Wunsafe-loop-optimizations]
  while(ev) {
       ^
events.c: In function ‘janus_events_type_to_name’:
events.c:381:7: warning: missed loop optimization, the loop counter may overflow [-Wunsafe-loop-optimizations]
  while(ev) {
       ^
```

To reproduce, build with e.g., gcc 7.3.0 and -O1.

Not super high priority as I don't think `ev` can actually overrun here, but maybe a different loop invariant should be used.

--- a/events.c
+++ b/events.c
@@ -29,7 +29,8 @@ static struct janus_event_types {
 	{ JANUS_EVENT_TYPE_MEDIA, "media", "Media"},
 	{ JANUS_EVENT_TYPE_PLUGIN, "plugins", "Plugins"},
 	{ JANUS_EVENT_TYPE_TRANSPORT, "transports", "Transports"},
-	{ JANUS_EVENT_TYPE_CORE, "core", "Core"}
+	{ JANUS_EVENT_TYPE_CORE, "core", "Core"},
+	{ -1, NULL, NULL},
 };
 
 static gboolean eventsenabled = FALSE;
@@ -346,7 +347,7 @@ void janus_events_edit_events_mask(const char *list, janus_flags *target) {
 						index++;
 					if(strlen(index)) {
 						struct janus_event_types *ev = event_types_string;
-						while(ev) {
+						while(ev->label) {
 							if(!strcasecmp(index, ev->label)) {
 								janus_flags_set(&mask, ev->type);
 								break;
@@ -368,7 +369,7 @@ void janus_events_edit_events_mask(const char *list, janus_flags *target) {
 /* Helpers to convert an event type to a string label or a more verbose name */
 const char *janus_events_type_to_label(int type) {
 	struct janus_event_types *ev = event_types_string;
-	while(ev) {
+	while(ev->label) {
 		if(type == ev->type)
 			return ev->label;
 		ev++;
@@ -378,7 +379,7 @@ const char *janus_events_type_to_label(int type) {
 
 const char *janus_events_type_to_name(int type) {
 	struct janus_event_types *ev = event_types_string;
-	while(ev) {
+	while(ev->label) {
 		if(type == ev->type)
 			return ev->name;
 		ev++;

